src/main/java/org/reactome/addlinks/fileprocessors/MGIFileProcessor.java

```diff
@@ -0,0 +1,52 @@
+package org.reactome.addlinks.fileprocessors;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class MGIFileProcessor extends FileProcessor{
+
+    public MGIFileProcessor(String processorName)
+    {
+        super(processorName);
+    }
+
+    public MGIFileProcessor()
+    {
+        super(null);
+    }
+
+    @Override
+    public Map<String, List<String>> getIdMappingsFromFile()
+    {
+        Map<String, List<String>> mappings = new HashMap<>();
+        Path inputFile = Paths.get(this.pathToFile.toAbsolutePath().toString().replace(".gz", ""));
+        try
+        {
+            for (String line : Files.readAllLines(inputFile).stream().collect(Collectors.toList())) {
```

Comment:
Streaming and collecting here can be removed:

```
for (String line : Files.readAllLines(inputFile)) {
...
}
```

should be equivalent.

Reply:
Also, you can do 
```Java
Files.readAllLines(inputFile).stream().forEach(line -> {
    ...
});
```

But this probably isn't worth it, unless you can safely use `parallelStream`. But even then, you probably won't notice much difference in performance unless the file is very large, or the per-line computation is very complex, or both. ...so maybe don't do that. ;)

Reply:
I admit, I may have been a little to `streams`-happy in the past. :(

Reply:
It's okay, I'm also starting to rebalance from my `streams` happiness :).  I do still really like them for pure functional transformation/filtering operations, though.

Reply:
They do _look_ nicer, but often under-perform, unless they are working on something that can be parallelized. That being said, if the input is small, the loss in performance is probably not _too_ noticeable.

Reply:
Yeah, for clarity I like to use them for small/medium-sized input, but I'd be willing to scrap it for the traditional loop if it's a bottleneck.

I do prefer to go with the clearest way to write the code first, though, and then optimize after doing test runs.

Reply:
OK made that change, but for the cases where I needed to skip the header the stream was still required, so for that it stayed